src/test/java/firstGroup/Assignment1WithBase.java

```diff
@@ -0,0 +1,410 @@
+package firstGroup;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import static com.codeborne.selenide.WebDriverRunner.getWebDriver;
+
+public class Assignment1WithBase extends BaseTest{
+
+//********With base*************************
+
+    @Test
+    public void test1(){
+        System.out.println("Thread Id: " +  Thread.currentThread().getId());
+        getDriver().navigate().to("https://www.flipkart.com/");
+        startOperationBegin();
+        System.out.println("test1 = ");
+        Boolean b =   mainOperation();
+        Assert.assertTrue(b);
+    }
+    @Test
+    public void test2(){
+        System.out.println("Thread Id: " +  Thread.currentThread().getId());
+        getDriver().navigate().to("https://www.flipkart.com/");
+        startOperationBegin();
+        System.out.println("test2 = ");
+        Boolean b =   mainOperation();
+        Assert.assertTrue(b);
+    }
+    //***************Validate Add to Cart Functionality***************
+    @Test
+    public void test3(){
+        System.out.println("Thread Id: " +  Thread.currentThread().getId());
+        getDriver().navigate().to("https://www.flipkart.com/");
+        startOperationBegin();
+        System.out.println("Done startOperationBegin");
+        addProduct();
+
+    }
+
+    public void addProduct(){
+
+        Assignment1Definition obj1 = new Assignment1Definition(driver.get());
+        try {
+            Thread.sleep(3000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        String shoeOneTextFromList = obj1.getTextOfShoeFirst().getText();
+        System.out.println("shoeOneTextFromList = "+shoeOneTextFromList);
+        obj1.clickFirstShoe().click();
+
+//        Set<String> windows =	driver.getWindowHandles();
+//Thread.sleep
+
+        Set<String> windows =   driver.get().getWindowHandles();
+
+        Iterator<String> itrWindows = windows.iterator();
```

Comment:
instead of defining a Set windows, use :
Iterator<String> itrWindows = driver.get().getWindowHandles().iterator();